utils/utils.go

```diff
@@ -111,13 +111,11 @@ func DownloadFile(ctx context.Context, rawURL string, logger logging.Logger) (ou
 	}
 
 	logger.Infof("Starting download of %s", rawURL)
-
 	parsedPath := parsedURL.Path
-	if runtime.GOOS == "windows" && !strings.HasSuffix(parsedPath, ".exe") {
-		parsedPath += ".exe"
-	}
-
 	outPath = filepath.Join(ViamDirs["cache"], path.Base(parsedPath))
+	if runtime.GOOS == "windows" && !strings.HasSuffix(outPath, ".exe") {
+		outPath += ".exe"
+	}
```

Comment:
why move this down so it applies to `outPath` but not `parsedPath`?

Reply:
Because of the `file://` handler below. We'll use the original parsedPath to open and copy the source file. If we appened .exe to it, now it's an invalid filename that doesn't exist. (Hit this in windows testing.)